R/sugar.R

```diff
@@ -66,3 +66,109 @@ opt = function(.key, ...) {
 opts = function(.keys, ...) {
   dictionary_sugar_mget(mlr_optimizers, .keys, ...)
 }
+
+#' @title Black-Box Optimization
+#'
+#' @description
+#' This function optimizes a function or [Objective] with a given method.
+#' 
+#' @param x (`function` | [Objective]).
+#' @param method (`character(1)`)\cr
+#'  Key to retrieve optimizer from [mlr_optimizers] dictionary.
+#' @param max_evals (`integer(1)`)\cr
+#'  Number of allowed evaluations.
+#' @param max_time (`integer(1)`)\cr
+#'  Maximum allowed time in seconds.
+#' @param lower (`numeric()`)\cr
+#'  Lower bounds on the parameters. If named, names are used to create the
+#'  domain.
+#' @param upper (`numeric()`)\cr
+#'  Upper bounds on the parameters.
+#' @param maximize (`logical()`)\cr
+#'  Logical vector used to create the codomain e.g. c(TRUE, FALSE) -> ps(y1 = p_dbl(tags = "maximize"), y2 = pd_dbl(tags = "minimize")).
+#'  If named, names are used to create the codomain.
+#' @param search_space ([paradox::ParamSet]).
+#' @param ... (named `list()`)\cr
+#'  Named arguments to be set as parameters of the optimizer.
+#' 
+#' @note
+#' If both `max_evals` and `max_time` are `NULL`, [TerminatorNone] is used. 
+#' This is useful if the [Optimizer] can terminate itself. 
+#' If both are given, [TerminatorCombo] is created and the optimization stops
+#' if the time or evaluation budget is exhausted.
+#'
+#' @return list of
+#'  * `par` - Best found parameters
+#'  * `value` - Optimal outcome
+#'  * `instance` - [OptimInstanceSingleCrit]
+#'
+#' @export
+#' @examples
+#' # function and bounds
+#' fun = function(xs) {
+#'   - (xs[[1]] - 2)^2 - (xs[[2]] + 3)^2 + 10
+#' }
+#'
+#' bb_optimize(fun, lower = c(-10, -5), upper = c(10, 5), max_evals = 10)
+#'
+#' # objective
+#' library(paradox)
+#' 
+#' fun = function(xs) {
+#'   c(z = - (xs[[1]] - 2)^2 - (xs[[2]] + 3)^2 + 10)
+#' }
+#' 
+#' domain = ps(x1 = p_dbl(-10, 10), x2 = p_dbl(-5, 5))
+#' codomain = ps(z = p_dbl(tags = "minimize"))
+#' objective = ObjectiveRFun$new(fun, domain, codomain)
+#'
+#' bb_optimize(objective, method = "random_search", max_evals = 10)
+#' 
+#' 
+bb_optimize = function(x, method = "random_search", max_evals = 100, max_time = NULL, ...) {
+  assert_int(max_time, lower = 0, null.ok = TRUE)
+  assert_int(max_evals, lower = 0, null.ok = TRUE)
+  assert_choice(method, mlr_optimizers$keys())
```

Comment:
I would go with @mb706's approach here and let method also be a constructed optimizer.